tools/wptserve/wptserve/response.py

```diff
@@ -183,6 +183,9 @@ def iter_content(self, read_file=False):
         True, the entire content of the file will be returned as a string facilitating
         non-streaming operations like template substitution.
         """
+        if not isinstance(self.content, Iterable) and not hasattr(self.content, "read"):
+            raise TypeError("Response content must be an iterable or file-like object!")
```

Comment:
Raising an exception here doesn't work; that only works inside the `handler()` call.

```
________________ TestFunctionHandler.test_non_iterable_content _________________
self = <functional.test_handlers.TestFunctionHandler testMethod=test_non_iterable_content>
    def test_non_iterable_content(self):
        @wptserve.handlers.handler
        def handler(request, response):
            response.content = 1234
            return response
    
        route = ("GET", "/non_iterable_rv", handler)
        self.server.router.register(*route)
        resp = self.request(route[1])
>       self.assertEqual(500, resp.getcode())
E       AssertionError: 500 != 200
wptserve/tests/functional/test_handlers.py:205: AssertionError
----------------------------- Captured stderr call -----------------------------
DEBUG:web-platform-tests:Route pattern: ^/(.*)$
DEBUG:web-platform-tests:Route pattern: ^/(.*\.asis)$
DEBUG:web-platform-tests:Route pattern: ^/(.*\.py)$
DEBUG:web-platform-tests:Using default configuration
INFO:web-platform-tests:Starting http server on localhost:35117
DEBUG:web-platform-tests:Route pattern: ^/non\_iterable\_rv$
DEBUG:web-platform-tests:GET /non_iterable_rv
DEBUG:web-platform-tests:Found handler FunctionHandler
DEBUG:web-platform-tests:200 GET /non_iterable_rv (None) 0
ERROR:web-platform-tests:Traceback (most recent call last):
  File "/home/travis/build/web-platform-tests/wpt/tools/wptserve/wptserve/server.py", line 544, in handle_one_request
    self.finish_handling_h1(request_line_is_valid)
  File "/home/travis/build/web-platform-tests/wpt/tools/wptserve/wptserve/server.py", line 243, in finish_handling_h1
    self.finish_handling(request, response, handler)
  File "/home/travis/build/web-platform-tests/wpt/tools/wptserve/wptserve/server.py", line 278, in finish_handling
    response.write()
  File "/home/travis/build/web-platform-tests/wpt/tools/wptserve/wptserve/response.py", line 223, in write
    self.write_content()
  File "/home/travis/build/web-platform-tests/wpt/tools/wptserve/wptserve/response.py", line 217, in write_content
    for item in self.iter_content():
  File "/home/travis/build/web-platform-tests/wpt/tools/wptserve/wptserve/response.py", line 187, in iter_content
    raise TypeError("Response content must be an iterable or file-like object!")
TypeError: Response content must be an iterable or file-like object!
```

Reply:
What I thought was any non-interable response content will raise a `TypeError` and an HTTP request gets a `500` status error. I assigned a number to `response.content` in handler on purpose. But obviously, it didn't work. Any good solution suggestion?

Sorry for this PR that didn't pass the unit test. I failed to set up CI environment before. My local CI environment for tools unit tests is set up successfully now. I'll run more unit test before I file any PRs.